src/Ledger/Transaction.lagda

```diff
@@ -73,7 +73,15 @@ the transaction body are:
   open EpochStructure epochStructure public
   open Ledger.Script crypto epochStructure public
 
-  field scriptStructure : ScriptStructure
+  field p1s : Hashable Timelock ScriptHash
+  field ps : PlutusStructure
+  field hashRespectsUnion : {A B Hash : Set} → Hashable A Hash → Hashable B Hash → Hashable (A ⊎ B) Hash
+
+  scriptStructure : ScriptStructure
+  scriptStructure = record { p1s = P1ScriptStructure-TL p1s ; 
+                              ps = ps ;
+                              hashRespectsUnion = hashRespectsUnion }
```

Comment:
This seems non-optimal; maybe deriving the script structure from `Hashable Timelock Scripthash` should have been added somewhere else, so that the general structure is not compromised.

Reply:
Fixed in commit 2193c12.